decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDecider.java

```diff
@@ -597,19 +627,79 @@ public void setConfig(Map<String, String> config) {
                 break;
             }
 
-            Bcl2FastqData data = new Bcl2FastqData(
-                    Iterables.getOnlyElement(laneNameToLaneProvenance.get(laneName)),
-                    laneNameToSampleProvenance.get(laneName));
-            data.setProperties(ImmutableMap.of("output_prefix", outputPath, "output_dir", outputFolder));
-            data.setMetadataWriteback(getDoMetadataWriteback());
-            data.setStudyToOutputPathConfig(studyToOutputPathConfig);
+            List<ProvenanceWithProvider<SampleProvenance>> samples = laneNameToSampleProvenance.get(laneName);
+            if (overrideBasesMask != null) {
+                try {
+                    samples = applyBasesMask(samples, overrideBasesMask);
+                } catch (DataMismatchException ex) {
+                    addError("Error while generating workflow run for lane = [{0}], errors:\n{1}", laneName, ex.toString());
+                    continue;
+                }
+            }
 
-            WorkflowRunV2 wr = handler.getWorkflowRun(metadata, data, getDoCreateIusLimsKeys() && !getIsDryRunMode(), getIsDemultiplexSingleSampleMode());
-            if (wr.getErrors().isEmpty()) {
-                validWorkflowRuns.add(wr);
-            } else {
-                invalidWorkflowRuns.add(wr);
-                log.error("Error while generating workflow run for lane = [{}], errors:\n{}", laneName, Joiner.on("\n").join(wr.getErrors()));
+            //get all barcodes in lane
+            Set<String> unsupportedBarcodeStrings = new HashSet<>();
+            List<Barcode> laneBarcodes = samples.stream()
+                    .map(ProvenanceWithProvider::getProvenance)
+                    .map(SampleProvenance::getIusTag)
+                    .map(barcodeString -> {
+                        try {
+                            return Barcode.fromString(barcodeString);
+                        } catch (DataMismatchException ex) {
+                            unsupportedBarcodeStrings.add(barcodeString);
+                            return null;
+                        }
+                    })
+                    .collect(Collectors.toList());
+            if (!unsupportedBarcodeStrings.isEmpty()) {
+                addError("Error while generating workflow run for lane = [{0}], unsupported barcode(s):\n{1}", laneName, unsupportedBarcodeStrings.toString());
+                continue;
+            }
+
+            //check for barcode collisions
+            List<Pair<Barcode, Barcode>> collisions = Barcodes.getCollisions(laneBarcodes, 1);
+            if (collisions.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)